packages/api-v2/src/hooks/p2p/entity/order/p2p-order/useOrderList.ts

```diff
@@ -8,30 +9,53 @@ const useOrderList = (
     config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_order_list'>>[1]>['options']
 ) => {
     const { isSuccess } = useAuthorize();
-    const { data, fetchNextPage, ...rest } = useInfiniteQuery('p2p_order_list', {
+    const { data: subscriptionData, subscribe, unsubscribe } = useSubscription('p2p_order_list');
+
+    // Subscribe to the p2p_order_list endpoint to keep track of the order list updates
+    React.useEffect(() => {
+        if (isSuccess) subscribe();
+
+        return () => {
+            unsubscribe();
+        };
+    }, [isSuccess]);
+
+    // Fetch the order list data which handles pagination
+    const {
+        data: queryData,
+        fetchNextPage,
+        refetch,
+        ...rest
+    } = useInfiniteQuery('p2p_order_list', {
         payload: { ...payload, offset: payload?.offset, limit: payload?.limit },
         options: {
             getNextPageParam: (lastPage, pages) => {
-                if (!lastPage?.p2p_order_list?.list?.length) return;
-
+                if (lastPage?.p2p_order_list?.list?.length === 0) return;
                 return pages.length;
             },
-            enabled: isSuccess && (config?.enabled === undefined || config.enabled),
+            enabled: subscriptionData && (config?.enabled === undefined || config.enabled),
         },
     });
 
+    // Refetch the data when the subscription data changes
+    React.useEffect(() => {
+        if (subscriptionData) {
+            refetch();
+        }
+    }, [subscriptionData, refetch]);
+
     // Flatten the data array
-    const flatten_data = React.useMemo(() => {
-        if (!data?.pages?.length) return;
+    const flattenedData = React.useMemo(() => {
```

Comment:
for the hooks variables aren't we using snake case itself?

Reply:
ah yeah true, ill rename these 😅 habit now 🤣